src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************\
+
+ Module: Generate Java Generic Type - Instantiate a generic class with
+         concrete type information.
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+#include "generate_java_generic_type.h"
+#include <util/namespace.h>
+#include <java_bytecode/java_types.h>
+#include <java_bytecode/java_utils.h>
+
+#include <iostream>
+#include <sstream>
+
+generate_java_generic_typet::generate_java_generic_typet(
+  symbol_tablet &symbol_table,
```

Comment:
As suggested - since the symbol table is mutable, please add as a parameter to the `operator()` allowing it to become `const`.